pkg/virt-launcher/virtwrap/cmd-server/server.go

```diff
@@ -304,6 +306,16 @@ func (l *Launcher) GetDomain(_ context.Context, _ *cmdv1.EmptyRequest) (*cmdv1.D
 	return response, nil
 }
 
+func decorateDomain(l *Launcher, domain *api.Domain) *api.Domain {
```

Comment:
I think it will be nicer to limit the use of mutable functions like this.
Instead of this one, you can directly set the OSInfo data and create a function that will fetch the interfaces data.